File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        // map<int,int>mp;//sum,index
        int sum=0;
        int n=nums.size();

        vector<int>prefix(n+1,0);

        for(int i=0;i<n;i++){
            prefix[i]=sum;
            sum+=nums[i];
        }

       prefix[n]=sum;
        int ans=0;
        for(int i=0;i<=n;i++){
            for(int j=i+1;j<=n;j++){
                if((prefix[j]-prefix[i])==k) ans++;
            }
        }
        return ans;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        dicti={}
        summ=0
        ans=0
        dicti[0]=1
        for val in nums:
            summ+=val
            target=summ-k
            if target in dicti:
                ans+=dicti[target]
            dicti[summ]=dicti.get(summ,0)+1
        return  ans